test/integration/private_zonal_with_networking/private_zonal_with_networking_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package private_zonal_with_networking
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/golden"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPrivateZonalWithNetworking(t *testing.T) {
+	bpt := tft.NewTFBlueprintTest(t)
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		// Skipping Test Due to location_policy Under AutoScaling in Node Pool for default-node-pool
+		// bpt.DefaultVerify(assert)
+
+		projectId := bpt.GetStringOutput("project_id")
+		location := bpt.GetStringOutput("location")
+		clusterName := bpt.GetStringOutput("cluster_name")
+		subnetName := bpt.GetStringOutput("subnet_name")
+		region := bpt.GetStringOutput("region")
+		ipRangePodsName := bpt.GetStringOutput("ip_range_pods_name")
+		ipRangeServicesName := bpt.GetStringOutput("ip_range_services_name")
+		serviceAccount := bpt.GetStringOutput("service_account")
+
+		op := gcloud.Runf(t, "container clusters describe %s --zone %s --project %s", clusterName, location, projectId)
+		g := golden.NewOrUpdate(t, op.String(),
+			golden.WithSanitizer(golden.StringSanitizer(projectId, "PROJECT_ID")),
+			golden.WithSanitizer(golden.StringSanitizer(clusterName, "CLUSTER_NAME")),
+			golden.WithSanitizer(golden.StringSanitizer(serviceAccount, "SERVICE_ACCOUNT")),
```

Comment:
Looks like the service account random suffix is causing is a diff. I wonder if its because the project ID sanitizer causes the service account project id part to be sanitized resulting in the `serviceAccount` value not matching. Could you try re arranging this so that `golden.WithSanitizer(golden.StringSanitizer(serviceAccount, "SERVICE_ACCOUNT"))` is the first followed by others?

Reply:
Modified and retriggred the Build

Reply:
Fixed the Service Account issue with reshuffling the Sanitising order

Reply:
The Golden File was updated with Sanitised SA but expected still shows up as the new SA 
even though we have it in the code

Reply:
The issue was Due to projectID being set the default Sanitiser Sanitised the SA for ProjectID and the remainder SA still existed. Fixed the same by unsetting ProjectID